### ISSUE TITLE
RabbitMQ Scalers -issues in connecting with TLS enabled 

### ISSUE THREAD

**Issue by maverickmetro** [kedacore/keda#967]
When we have a TLS enabled Rabbit MQ service hosted in the cloud, Rabbit MQ scaler is not able to connect to the Rabbit MQ host Receiving the below error in Keda Operator API service pod . In Rabbit MQ Scaler i used the apiHost parameter with https protocol. 

```
E0803 23:09:48.173311       1 provider.go:94] keda_metrics_adapter/provider "msg"="error getting metric for scaler" "error"="error inspecting rabbitMQ: Get \"https://ibm_cloud_26d26b44_4a05_4a02_a7fd_3423c5f6a4f7:***@a5adefe4-8087-450a-83c9-113f3de3b9b0.8117147f814b4b2ea643610826cd2046.databases.appdomain.cloud:30347/api/queues/%2F/KedaTestQueue\": x509: certificate signed by unknown authority"  "ScaledObject.Name"="keda-sample" "ScaledObject.Namespace"="default" "Scaler"={}
E0803 23:09:48.173371       1 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"No matching metrics found for queuelength"}
```

## Steps to Reproduce the Problem

  1. Install Rabbit MQ with TLS enabled
  2. Install Keda 
  3. Configure Rabbit MQ scaler scaled object to use apiHost Parameter with https protocol.

 Sample Scaled Object Yaml:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: rabbitmq-consumer-secret
data:
  RabbitHttpHost: xxxxxxxxxxx #https://XXXXXXXXXXXXX
---
apiVersion: keda.k8s.io/v1alpha1
kind: ScaledObject
metadata:
  name: keda-sample
  namespace: default
  labels:
    deploymentName: keda-sample
spec:
  scaleTargetRef:
    deploymentName: keda-sample
  pollingInterval: 5 # Optional. Default: 30 seconds
  cooldownPeriod: 30 # Optional. Default: 300 seconds
  minReplicaCount: 1
  maxReplicaCount: 5 # Optional. Default: 100
  triggers:
    - type: rabbitmq
      metadata:
        includeUnacked: "true"
        queueName: KedaTestQueue
        queueLength: "5"
        
      authenticationRef:
        name: rabbitmq-consumer-trigger
---
apiVersion: keda.k8s.io/v1alpha1
kind: TriggerAuthentication
metadata:
  name: rabbitmq-consumer-trigger
  namespace: default
spec:
  secretTargetRef:
    - parameter: apiHost
      name: rabbitmq-consumer-secret
      key: RabbitHttpHost

---
```

## Specifications

  - **KEDA Version:** 1.5.0
  - **Platform & Version:** IBM public Cloud
  - **Kubernetes Version:** 1.15.12
  - **Scaler(s):** RabbitMQ Scaler

// EDIT: fixed formating (Zbynek)

**Comment by zroubalik** [comment]
I have done a quick check and RabbitMQ Scaler TLS support is not implemented yet. https://github.com/kedacore/keda/blob/v2/pkg/scalers/rabbitmq_scaler.go#L138 this would need to use `amq.DialTLS()`. And consume the ceritficate the similar way it does for example for Kafka Scaler.

@maverickmetro is this something you are willing to contribute?

**Comment by ansonauhk** [comment]
Please review, I submitted a fix 
code - https://github.com/kedacore/keda/pull/1051 
doc - https://github.com/kedacore/keda-docs/pull/240

**Comment by mohite-abhi** [comment]
I have added tls support for RabbitMQ getting inspiration from [above MR](https://github.com/kedacore/keda/issues/967#issuecomment-682334742), it's working for me. I would like to add it with Keda code. Can we reopen this issue, or should I create a new issue regarding it?

**Comment by mohite-abhi** [comment]
this issue is tracking some other closed pr when it should have been closed by #4086 merge.
